### PR TITLE
support multiple mutators

### PR DESCRIPTION
**Pull request by jakkab**
## Related issue
implements #233 

## Proposed changes
Support multiple mutators

## Checklist
- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md)
- [x] I have read the [security policy](../security/policy)
- [x] I confirm that this pull request does not address a security vulnerability. If this pull request addresses a security
vulnerability, I confirm that I got green light (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation within the code base (if appropriate)
- [x] I have documented my changes in the [developer guide](https://github.com/ory/docs) (if appropriate)

### CODE REVIEW

**Comment by aeneasr** [comment]
😎

**Comment by aeneasr** [comment]
LMK when good for 👀

File: rule/validator.go [hunk 1]
```diff
@@ -95,17 +95,24 @@ func (v *ValidatorDefault) validateAuthorizer(r *Rule) error {
...
-		return errors.WithStack(herodot.ErrInternalServerError.WithReason(`Value of "mutator.handler" can not be empty.`))
+func (v *ValidatorDefault) validateMutators(r *Rule) error {
+	if len(r.Mutators) == 0 {
+		return errors.WithStack(herodot.ErrInternalServerError.WithReason(`Value of "mutators" must be set and can not be an empty array.`))
 	}
 
-	mutator, err := v.r.PipelineMutator(r.Mutator.Handler)
-	if err != nil {
-		return herodot.ErrInternalServerError.WithReasonf(`Value "%s" of "mutator.handler" is not in list of supported mutators: %v`, r.Mutator.Handler, v.r.AvailablePipelineMutators()).WithTrace(err).WithDebug(err.Error())
+	for k, m := range r.Mutators {
```
Comment (piotrmsc) [inline rule/validator.go]: Maybe we should return a list of wrong mutators instead of returning error for single element?

**Comment by jakkab** [comment]
@aeneasr ready for review!

File: pipeline/mutate/mutator_cookie.go [hunk 2]
```diff
@@ -91,7 +93,9 @@ func (a *MutatorCookie) Mutate(r *http.Request, session *authn.AuthenticationSes
 		}
 	}
 
-	return req.Header, nil
+	session.SetHeader(cookieHeader, req.Header.Get(cookieHeader))
```
Comment (aeneasr) [inline pipeline/mutate/mutator_cookie.go]: That makes a lot of sense!

File: proxy/request_handler.go [hunk 3]
```diff
@@ -156,31 +156,49 @@ func (d *RequestHandler) HandleRequest(r *http.Request, rl *rule.Rule) (http.Hea
...
+			return nil, err
+		}
+
+		if err := sh.Mutate(r, session, m.Config, rl); err != nil {
+			d.r.Logger().WithError(err).
+				WithField("granted", false).
+				WithField("access_url", r.URL.String()).
+				WithField("authorization_handler", m.Handler).
+				WithField("reason_id", "mutation_handler_error").
+				Warn("The authorization handler encountered an error")
```
Comment (aeneasr) [inline proxy/request_handler.go]: The mutator handler :)

Comment (jakkab) [reply proxy/request_handler.go]: copy-paste backfired :<

**Comment by aeneasr** [comment]
🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉

**Pull request merged**